packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -5007,7 +5012,17 @@ export default class Meeting extends StatelessWebexPlugin {
 
       this.roap.abortTurnDiscovery();
 
-      if (joined && isRetry) {
+      // if this was the first attempt, let's do a retry
+      let shouldRetry = !isRetry;
+
+      if (CallDiagnosticUtils.isSdpOfferCreationError(error)) {
+        // errors related to offer creation (for example missing H264 codec) will happen again no matter how many times we try,
+        // so there is no point doing a retry
+        shouldRetry = false;
+      }
+
+      // we only want to call leave if join was successful and this was a retry or we won't be doing any more retries
+      if (joined && (isRetry || !shouldRetry)) {
```

Comment:
These changes here are not strictly required for this fix, but it's something I've found while working on this and initially I thought I would need this fix, but later changed my approach. The issue here was that if joinWithMedia() failed and decided that there is no point in doing a retry, it wasn't calling leave() and it should, so I've fixed it.